### DISCUSSION TITLE
People Search - Sort settings title

### DISCUSSION THREAD

**Discussion by gitulm** [microsoft-search/pnp-modern-search#2159] [General]
When searching for a person, the name of the person is indicated by the title. When I try to sort by it, I get an error message:

![image](https://user-images.githubusercontent.com/56668705/174569278-32ef843c-ecdf-4a28-be47-3632e3d51a64.png)

What am I doing wrong? How to sort people's names alphabetically?

**Comment by wobba** [comment]
If you want to sort on names you need to map Title/Firstname/Lastname of the crawled properties for the profile to a RefinableString managed property, and then use that for sorting. This old post (https://sympmarc.com/2016/06/09/create-a-simple-sharepoint-2013-employee-directory-on-office365-part-5-sorting-refiners/) from @sympmarc highlights what is needed.

**Reply by gitulm** [reply]
Thanks for the hint. However, I still have a request. When searching for a person, I would like to see the result sorted by last name and first name. When I look in debug mode, it says it's the "Title". Looking in the UserProfileService, the field is "PreferredName" for displaying last name, first name. Which "Title" or "PreferredName" should I assign a RefinableString to? Since the "Title" field is displayed when searching for people, which "Title" should I connect to a RefinableString? Can someone tell me the exact table?

**Reply by wobba** [reply]
If you want to sort on lastname, you need to map a crawled property to a managed property, and sort on that one. E.g. map `People:PreferredName` to `RefinableString100`. Then you need to trigger re-indexing and wait up to 48h. See https://github.com/wobba/SPO-Trigger-Reindex

**Reply by agilbull** [reply]
Thank you for this answer. Unfortunately, I tried to map several properties to RefinableStringXX in order to sort. But all RefinableStrings are empty. I get no data. I waited one week but still no data

Here is debug view
"AuthorOWSUSER": null,
        "contentclass": "urn:content-class:SPSPeople",
        "ContentTypeId": null,
        "Created": null,
        "CreatedBy": null,
        "DefaultEncodingURL": null,
        "EMailOWSTEXT": null,
        "FileType": null,
        "HitHighlightedSummary": null,
        "JobTitle": "MME",
        "NormListID": null,
        "NormSiteID": null,
        "NormUniqueID": null,
        "NormWebID": null,
        "owstaxidmetadataalltagsinfo": null,
        "Path": "https://cnavts-my.sharepoint.com/Person.aspx?accountname=i:0%23.f|membership|xxxxxxxxxxxxxxxxx",
        "People": null,
        "SiteTitle": null,
        "SPSiteUrl": null,
        "Title": "XXXXXXXXX xxxxxxx",
        "UniqueID": "{00000000-0000-0000-0000-000000000000}",
        "UserName": "xxxxxxxx@XXXXXXXX",
        "WorkPhone": "0000000000",
        "RefinableString01": null,
        "RefinableString02": null,
        "RefinableString03": null,
        "RefinableString04": null,
        "RefinableString05": null,
        "RefinableString06": null,
        "RefinableString07": null,
        "RefinableString08": null,
        "RefinableString10": null,
        "RefinableString09": null,

![image](https://user-images.githubusercontent.com/26926003/193782617-188bcfef-8ad0-47f2-bd2a-4b6b4784f963.png)


Any idea?

**Reply by wobba** [reply]
You need to re-index data to populate the properties. Check out https://pnp.github.io/script-samples/spo-request-pnp-reindex-user-profile/README.html

**Reply by agilbull** [reply]
Thank you very much, I'll try it.
